src/proxy.js

```diff
@@ -94,93 +190,178 @@ arrayTraps.set = function(state, prop, value) {
 	return objectTraps.set.call(this, state[0], prop, value)
 }
 
-// returns the object we should be reading the current value from, which is base, until some change has been made
-function source(state) {
+// Used by Map and Set drafts
+const reflectTraps = makeReflectTraps([
+	"ownKeys",
+	"has",
+	"set",
+	"deleteProperty",
+	"defineProperty",
+	"getOwnPropertyDescriptor",
+	"preventExtensions",
+	"isExtensible",
+	"getPrototypeOf"
+])
+
+/**
+ * Map drafts
+ */
+
+const mapTraps = makeTrapsForGetters({
```

Comment:
Random thought: could we use the ES5 implementation, and not use Proxies for maps and sets at all? If that would simplify the implementation that would be cool. If "ES5" administration doesn't play nicely inside a proxy draft tree, than lets leave it as is. But proxies themself don't add much for maps and sets, since all object interactions are through functions anyway.

Reply:
Could also be opened as follow up optimization issue, to reduce the package size

Reply:
I think it's possible to use one codebase for Map and Set and use both Proxy and ES5 mechanics. Let's do it as a follow up issue

Reply:
See #449